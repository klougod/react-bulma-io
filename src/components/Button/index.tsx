import React, { ButtonHTMLAttributes, ReactNode } from 'react'

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  text?: ReactNode
  isLoading?: Boolean
  onClick?: () => void
}

export const Button = ({ className, text, isLoading, onClick }: ButtonProps) => (
  <button
    onClick={onClick}
    className={`button ${className} ${isLoading ? 'is-loading' : ''}`}
  >
    {text}
  </button>
)

Button.defaultProps = {
  isLoading: false,
  onClick: () => {},
  text: '',
  className: ''
}

export default Button
