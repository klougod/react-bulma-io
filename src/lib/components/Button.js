import React from 'react'
import PropTypes from 'prop-types'

export const Button = ({ className, text, isLoading, onClick }) => (
  <button onClick={onClick} className={`button ${className} ${isLoading && 'is-loading'}`}>
    {text}
  </button>
)

Button.defaultProps = {
  isLoading: false,
  onClick: null,
  text: '',
  className: ''
}

Button.propTypes = {
  isLoading: PropTypes.bool,
  onChange: PropTypes.func,
  text: PropTypes.string,
  className: PropTypes.string
}

export default Button