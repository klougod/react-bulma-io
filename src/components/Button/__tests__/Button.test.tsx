import React from 'react'
import Enzyme from 'enzyme'

import Button from '../'

describe('Button component test', function () {
  it('should render without crash with wrapper', function () {
    const buttonMock = Enzyme.shallow(<Button />)
    expect(buttonMock).toMatchSnapshot()
  })

  it('should render without loading', function () {
    const buttonMockProps = {
      text: 'button text',
      className: 'buttonClass',
      onClick: () => {},
      isLoading: false
    }
    const buttonMock = Enzyme.shallow(<Button {...buttonMockProps} />)
    expect(buttonMock).toMatchSnapshot()
  })

  it('should render loading', function () {
    const buttonMockProps = {
      text: 'button text',
      className: 'buttonClass',
      onClick: () => {},
      isLoading: true
    }
    const buttonMock = Enzyme.shallow(<Button {...buttonMockProps} />)
    expect(buttonMock).toMatchSnapshot()
  })
})
