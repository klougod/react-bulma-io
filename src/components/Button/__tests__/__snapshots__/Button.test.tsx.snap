// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test should render loading 1`] = `
<button
  className="button buttonClass is-loading"
  onClick={[Function]}
>
  button text
</button>
`;

exports[`Button component test should render without crash with wrapper 1`] = `
<button
  className="button  false"
  onClick={null}
/>
`;

exports[`Button component test should render without loading 1`] = `
<button
  className="button buttonClass false"
  onClick={[Function]}
>
  button text
</button>
`;
